tensorflow/python/training/slot_creator.py

```diff
@@ -85,11 +85,16 @@ def _create_slot_var(primary, val, scope, validate_shape, shape, dtype):
     # remove "'linear//weight' + '/'" and ':0'.
     real_slot_name = slot.name[len(primary.op.name + "/"):-2]
     slice_info = primary._save_slice_info
-    slot._set_save_slice_info(variables.Variable.SaveSliceInfo(
-        slice_info.full_name + "/" + real_slot_name,
-        slice_info.full_shape[:],
-        slice_info.var_offset[:],
-        slice_info.var_shape[:]))
+    # support slot's shape not same as primary's shape
+    # example: primary's shape = [10, 20, 30], slot's shape =
+    # [], [10], [10, 20] or [10, 20, 30] is allowed
+    n = slot.shape.ndims
+    if (n > 0):
```

Comment:
Should we still copy over the save_slice_info like before if ndims is None? Variables can have unknown shapes.

Also, I think the linter will complain about extra parentheses. Mind taking them out?

Otherwise looks great.

Reply:
1. Do you mean 'n' may be None? In TF, Variable rank is fixed, ndims is fixed, the dimension value may be 'unknown shapes'
2. I will fix 'extra parentheses'

Reply:
Variables can have an unknown number of dimensions:

```
v = tf.Variable(1., shape=tf.TensorShape(None))
assert v.shape.ndims is None
v.assign([[1.]])
```

The shape= argument is relatively new, but inside a tf.function it's been possible to make variables with relaxed shapes for a while.

Reply:
updated
thanks